unity-renderer/Assets/Scripts/MainScripts/DCL/AvatarSystem/AvatarEmotesController.cs

```diff
@@ -41,14 +41,25 @@ public void AddVisibilityConstraint(string key)
             visibilityConstraints.Add(key);
 
             if (!CanPlayEmote())
-                StopEmote();
+                StopEmote(true);
         }
 
         public void RemoveVisibilityConstraint(string key)
         {
             visibilityConstraints.Remove(key);
         }
 
+        public void UpdateEmoteStatus(string currentEmoteId, long timestamp)
```

Comment:
I feel like the naming of the function doesn't describe something meaningful, the status could be anything. Perhaps something like: `PlayOrStopEmote` ?